src/Nethermind/Nethermind.Optimism/CL/P2P/PayloadByNumberProtocol.cs

```diff
@@ -14,18 +14,30 @@
 
 namespace Nethermind.Optimism.CL.P2P;
 
-public class PayloadByNumberProtocol(ulong chainId, IPayloadDecoder payloadDecoder, ILogger logger)
-    : ISessionProtocol<ulong, ExecutionPayloadV3?>
+public class PayloadByNumberProtocol : ISessionProtocol<ulong, ExecutionPayloadV3?>
 {
-    public string Id => $"/opstack/req/payload_by_number/{chainId}/0";
+    private const int MaxResponseSizeBytes = 10000000;
+    private readonly ulong _chainId;
+    private readonly IPayloadDecoder _payloadDecoder;
+    private readonly ILogger _logger;
+
+    public string Id => $"/opstack/req/payload_by_number/{_chainId}/0";
```

Comment:
Is it called only once? Not sure how many formats it will incur.

Reply:
Not really sure since it's required by the `ISessionProtocol` interface. We don't make direct usage of this property.